rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -513,7 +575,18 @@ void deserialize_array<std::string>(
 {
     (void)call_new;
     if (member->array_size_ && !member->is_upper_bound_) {
-        deser.deserializeArray(((rosidl_generator_c__String*)field)->data, member->array_size_);
+        auto deser_field = (rosidl_generator_c__String*)field;
+        std::string stringseq[member->array_size_];
+        deser.deserializeArray(stringseq, member->array_size_);
+        for (size_t i = 0; i < member->array_size_; ++i) {
+          // NOTE: Do we need to call init ?
```

Comment:
It seems to me that it's not needed given that assign will call `rosidl_generator_c__String__fini` on the data structure before trying to assign elements. But I wasn't sure so I kept it commented hoping that someone would confirm it during review